wandb/sdk/launch/launch.py

```diff
@@ -1,32 +1,95 @@
 import logging
-from typing import Any, Dict, List, Optional
+import os
+from typing import Any, Dict, List, Optional, Tuple
 
 from wandb.apis.internal import Api
 from wandb.errors import ExecutionError
+import yaml
 
 from ._project_spec import create_project_from_spec, fetch_and_validate_project
 from .agent import LaunchAgent
+from .builder import loader as builder_loader
 from .runner import loader
 from .runner.abstract import AbstractRun
 from .utils import (
     construct_launch_spec,
+    LAUNCH_CONFIG_FILE,
     PROJECT_DOCKER_ARGS,
     PROJECT_SYNCHRONOUS,
+    resolve_build_and_registry_config,
 )
 
 _logger = logging.getLogger(__name__)
 
 
+def resolve_agent_config(
+    api: Api,
+    entity: Optional[str],
+    project: Optional[str],
+    max_jobs: Optional[float],
+    queues: Optional[List[str]],
+) -> Tuple[Dict[str, Any], Api]:
+    defaults = {
+        "entity": api.default_entity,
+        "max_jobs": 1,
+        "queues": ["default"],
+        "api_key": api.api_key,
+        "base_url": api.settings("base_url"),
+        "registry": {},
+        "build": {},
+    }
+
+    resolved_config: Dict[str, Any] = defaults
+    if os.path.exists(os.path.expanduser(LAUNCH_CONFIG_FILE)):
+        config = {}
+        with open(os.path.expanduser(LAUNCH_CONFIG_FILE)) as f:
+            try:
+                config = yaml.safe_load(f)
+                print(config)
+            except yaml.YAMLError as e:
+                print(e)
```

Comment:
Should this throw a wandb specific error here instead of just print?

Reply:
Nice catch.